src/javascript/components/SitemapPanelHeader/SitemapPanelHeader.jsx

```diff
@@ -4,21 +4,70 @@ import {Book, Button, Delete, Header, Save, Upload} from '@jahia/moonstone';
 import styles from './SitemapPanelHeader.scss';
 import {DialogComponent} from '../Dialog/Dialog';
 import {useTranslation} from 'react-i18next';
+import {useMutation} from '@apollo/react-hooks';
+
+import * as gqlMutations from './gqlMutations';
 
 export const SitemapPanelHeaderComponent = ({
     formik,
     isSitemapMixinEnabled,
-    siteKey
+    siteKey,
+    snackBarInfo,
+    openSnackBar
 }) => {
     const {t} = useTranslation('sitemap');
     const [dialogIsOpen, setDialogIsOpen] = useState(false);
     const [dialogInfo, setDialogInfo] = useState(null);
 
-    const handleDialogOpen = (title, text, submitText) => {
+    const [submitToGoogleMutation] = useMutation(gqlMutations.sendSitemapToSearchEngine, {
+        variables: {
+            sitemapURL: formik.values.sitemapIndexURL
+        },
+        // eslint-disable-next-line no-unused-vars
+        onCompleted: data => {
+            snackBarInfo(t('labels.snackbar.successSubmitToGoogle'));
+            openSnackBar(true);
+            handleDialogClose();
+        },
+        // eslint-disable-next-line no-unused-vars
+        onError: error => {
+            handleDialogClose();
+        }
+    });
+
+    const [deleteSitemapCacheMutation] = useMutation(gqlMutations.deleteSitemapCache, {
+        variables: {
+            expirationTimeDifference: formik.values.sitemapCacheDuration
+        },
+        // eslint-disable-next-line no-unused-vars
+        onCompleted: data => {
+            snackBarInfo(t('labels.snackbar.successFlushCache'));
+            openSnackBar(true);
+            handleDialogClose();
+        },
+        // eslint-disable-next-line no-unused-vars
+        onError: error => {
+            handleDialogClose();
+        }
+    });
+
+    const handleDialogOpen = (id, title, text, submitText) => {
+        let submitFunc = null;
```

Comment:
Do we maybe want to add a stub function or function that logs this exception instead of null? Not sure if DialogComponent handles null handleSubmit gracefully.

Reply:
Yup sounds good I will change it to
`() => {}`

Reply:
wouldn't hurt to also add `log.error('<unexpected error message here>')` in that function as well just in case :)

Reply:
Probably not good on the front-end side to log like that :)